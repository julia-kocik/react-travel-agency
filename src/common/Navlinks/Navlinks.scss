@import '../../settings.scss';

.navlinks_container {
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    .nav_container {
        height: 100%;
        width: 100%;
        color: $color-white;
        background-color: $color-blue;
        background: $color-gradient;
        background: $linear-gradient;
        list-style-type: none;
        display: flex;
        justify-content: center;
        align-items: center;
        margin: 0;
        font-size: 16px;
        .nav_items {
            margin: 0 30px;
            font-family: $font-main;
        }
    }
    .nav_links {
        text-decoration: none;
        color: $color-white;
    }
}
