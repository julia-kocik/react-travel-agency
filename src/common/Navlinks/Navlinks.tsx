import React from 'react'
import Navigation from '../../features/NavBar/NavItems/Navigation'
import './Navlinks.scss'

const Navlinks = () => {
  return (
    <div data-testid="navlinks-container" className='navlinks_container'>
      <Navigation/>
    </div>
  )
}

export default Navlinks