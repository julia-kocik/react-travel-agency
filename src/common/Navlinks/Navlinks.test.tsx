import React from 'react';
import { render } from '@testing-library/react';
import Navlinks from './Navlinks';
import './Navlinks.scss'

describe('Navlinks component', () => {
  it('should render correctly', () => {
    const { container } = render(<Navlinks />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should contain a div with class "navlinks_container"', () => {
    const { getByTestId } = render(<Navlinks />);
    expect(getByTestId('navlinks-container').classList.contains('navlinks_container')).toBe(true);
  });

  it('should contain a span with text "Julia Kocik 2023"', () => {
    const { getByText } = render(<Navlinks />);
    expect(getByText('Julia Kocik 2023')).toBeInTheDocument();
  });

});
