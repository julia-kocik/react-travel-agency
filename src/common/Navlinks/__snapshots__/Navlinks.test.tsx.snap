// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navlinks component should render correctly 1`] = `
<div
  class="navlinks_container"
  data-testid="navlinks-container"
>
  <span>
    Julia Kocik 2023
  </span>
</div>
`;
